core/src/main/java/hudson/model/listeners/RunListener.java

```diff
@@ -108,6 +109,12 @@ public void onCompleted(R r, @Nonnull TaskListener listener) {}
     public void onFinalized(R r) {}
 
     /**
+     * Called when a Run is entering execution.
+     * @param r
```

Comment:
it's not so much about the name - but it points out that this is a perfect thing to put in the documentation :-)